src/main/java/com/quizapp/enterprise/services/GameService.java

```diff
@@ -132,11 +151,11 @@ public void processPlayerTimeExpirationEvent(String playerName, String gameId) {
     }
 
     private void dispatchGameOrRoundOverEvents(GameStatus gameStatus, String gameCode){
-        if(gameStatus == GameStatus.GameEnded){
+        if(gameStatus.equals(GameStatus.GameEnded)){
             gameOverEventPublisher.publishGameOverEvent(gameCode, GameTracker.getInstance().getLeaderboard(gameCode));
             roundOverEventPublisher.publishRoundOverEvent(gameCode, GameTracker.getInstance().getLeaderboard(gameCode));
 
-        }else if(gameStatus == GameStatus.RoundEnded){
+        }else if(gameStatus.equals(GameStatus.RoundEnded)){
```

Comment:
change comparing convention to use the Object class's equals method rather than the "==" operator to provide better optimization and possible customized/specified "equality" in the future with override. 
This change was recommended by the "3. Object#equals Method" section of the "comparing objects in java" article referenced below

source: https://www.baeldung.com/java-comparing-objects#:~:text=Object%23equals%20Method&text=This%20method%20is%20defined%20in,same%20as%20the%20%3D%3D%20operator.

Reply:
Yeah... since it's an object, this is probably a good idea.